pkg/ocm/subscription/service.go

```diff
@@ -49,20 +53,64 @@ type SubscriptionGetter interface {
 func (client *subscriptionClient) GetSubscriptions() ([]Subscription, error) {
 	discovered := []Subscription{}
 	request := client.Config
-
 	request.Page = 1
+
+	logf.V(1).Info("Starting subscription retrieval", "BaseURL", client.Config.BaseURL, "Size", client.Config.Size)
+
 	for {
+		// Logging request details
+		logf.V(2).Info("Sending subscription request", "Page", request.Page, "Size", request.Size)
+
+		// Fetch the subscriptions
 		discoveredList, err := SubscriptionProvider.GetSubscriptions(request)
+
 		if err != nil {
-			return nil, fmt.Errorf(err.Error.Error())
+			if err.Error == nil && err.Reason != "" {
+				err.Error = fmt.Errorf(err.Reason)
+			}
+
+			logf.Error(err.Error, "Failed to retrieve subscriptions", "Page", request.Page,
+				"BaseURL", client.Config.BaseURL)
+
+			switch err.StatusCode {
+			case 401, 403:
+				logf.Info(fmt.Sprintf("%v error occurred, returning empty subscription list", err.StatusCode))
+				return []Subscription{}, nil
+
+			case 404:
+				logf.Info("404 error occurred, returning empty subscription list")
```

Comment:
Did you intend to have a return for the 404 case?

Reply:
Yes! Thank you for catching that. I've updated the code.